### DISCUSSION TITLE
Result cache, or git diff?

### DISCUSSION THREAD

**Discussion by freiondrej-lmc** [rectorphp/rector#6918] [Q&A] [answered]
Hello,
we are using Rector in our pipeline and it takes quite a long time to run. Therefore we want to examine possibilities to make it faster. We wanted to use git diff to run Rector only on the changed files, but we are not sure if there might be negative consequences, like those described in this Phpstan blogpost https://medium.com/@ondrejmirtes/from-minutes-to-seconds-massive-performance-gains-in-phpstan-163be88d1519 since Rector uses parts of Phpstan. We have also seen the PR https://github.com/rectorphp/rector/pull/3118 - does this mean we just have to configure our CI pipeline to use some shared cache and Rector will use it to store result cache files?
Thank you! Rector rocks 🚀

**Comment by freiondrej-lmc** [comment, marked as answer]
@TomasVotruba could you please point me in the right direction if you have some time? :)

**Reply by TomasVotruba** [reply]
Hi @freiondrej-lmc ,

I missed this one, I'm mostly in active issues :)
The link to PHPStan article is the full explanation why we have to analyze the whole project. But we use the cache for changed and dependent file from PHPStan.

That's why we always run full Rector on current code version.
As for CI speedup, depending on threads number. You can now re-use parallel run.

Maybe there is specific bottle neck you can improve, but I would have to see your exact project. E.g. skipping mPdf classes usually saved huge amount of time

**Reply by freiondrej-lmc** [reply]
Thank you! That makes sense, we'll implement the cache and try to take advantage of the parallel run you suggest. 🚀
